radix-engine/src/model/bucket.rs

```diff
@@ -152,7 +153,7 @@ impl Bucket {
         self.container.borrow_mut()
     }
 
-    pub fn main<S: SystemApi>(
+    pub fn main<S: SystemApi<W, I>, W: WasmEngine<I>, I: WasmInstance>(
```

Comment:
Bit of a code smell here in that `bucket.rs` needs to know about wasm

Reply:
Yep, was struggling a bit with Rust's generic type. The `Bucket` impl depends on `SystemApi` which is parameterized with `WasmEngine`.

Reply:
In the long term, we want to move away from compile-time-decided engine to runtime-decided engine, so multiple engines can coexist and there is not need for generic types (replaced with trait object).